### ISSUE TITLE
Support hard wrapping with newline characters

### ISSUE THREAD

**Issue by hkdobrev** [careteditor/caret#50]
What I mean by hard wrapping is instead of *just showing* the text as wrapped in the editor, *actually place newline characters* between words at around column 80. This way when the Markdown file is just typed normally in Caret, it would look the same when opened in an editor with wrapping disabled - e.g. Sublime Text by default, github.com file viewer, diffs, patches etc.

Vim supports this as an option and it is enabled by default when writing Git commits if you have installed Git properly and you are using Vim as your `$EDITOR`.

A lot of syntax and semantic details in Markdown come from email and email (and other wrapped text) is ideal when it is hand-wrapped. So Markdown was designed with this in mind so when there is a single newline character between words, to treat that as just a space and not put an actual new line in the rendered output unless there were either two spaces or two newline characters. So when typing I'd like to make use of that and instead of just visually wrapping to achieve hard wrapping automatically like I can with Vim.

P.S. Some thoughts from Linus Torvalds on manually wrapping text in emails: http://lxr.free-electrons.com/source/Documentation/email-clients.txt
P.P.S. Just to note this is really hard and probably not such a needed feature so I'm just putting it out there, but I don't expect it to be available anytime soon.

**Comment by erusev** [comment]
Good point, we've been discussing adding a command in the `Edit` menu next t o`Copy as HTML` that says sth like `Copy Wrapped` and adds this line breaks to the copied text.
